design/simpler-configuration.md

```diff
@@ -0,0 +1,267 @@
+      - name: sslmode
+        provider: literal
+        id: require
+  - name: http_good_basic_auth_handler
+    type: basic_auth
+    listener: http_default
+    match:
+      - ^http.*
+    credentials:
+      - name: username
+        provider: literal
+        id: someuser
+      - name: password
+        provider: literal
+        id: testpassword
+  - name: aws
+    listener: http_default
+    match:
+      - ".*"
+    credentials:
+      - name: accessKeyId
+        value:
+          environment: AWS_ACCESS_KEY_ID
+      - name: secretAccessKey
+        value:
+        environment: AWS_SECRET_ACCESS_KEY
+```
+
+Some of the key concerns we have with this configuration are:
+- It separates listeners and handlers conceptually, though we are moving toward combining these into a single `handler`
+  concept.
+- Having listeners and handlers separate in the configuration (where you have a `listeners` section and a `handlers`
+  section) also makes it harder to quickly parse the document to see what processing will be done on connections that come
+  in over a certain port or socket file. For example, the handler definition needs to refer to the listener name that it is
+  linked with.
+- The configuration is overly verbose:
+  - Both handlers and listeners must be named
+  - Rather than using keys as names, we require specifying the `name` of many things
+- The current configuration conflates credentials and configuration data
+- The http handler configuration is inconsistent over whether the name must match the http handler type or whether (as
+  is consistent with other handlers) you can use a random name and specify the `type`.
+- In expecting users to understand listeners and handlers to write a configuration, we're exposing an implementation detail
+  instead of crafting the configuration in terms that will be clear and sensible to the user.
+- In the current experience, we never took into account the user's mental model and designed our experience around that
+
+
+#### Proposed Experience
+```yaml
+version: "1"
+services:
+  postgres-db:
+    protocol: pg
+    listenOn: 0.0.0.0:5432 # can be a socket as well (same name for both)
+    credentials:
+      address: postgres.my-service.internal:5432
+      password:
+        providerId: name-in-vault
+        provider: vault
+      username:
+        providerId: username
+        provider: env
+    config:  # this section usually blank
+      optionalStuff: blah
+      
+  # the `aws` authStrategy specified in the config indicates which protocol implementation to use
+  aws-client:
+    protocol: http
+    listenOn: /var/docker/docker.sock
+    credentials:
+      accessKeyID:
+        providerId: name-in-vault
+        provider: conjur
+      secretAccessKey:
+        providerId: name-in-vault
+        provider: conjur
+      accessToken:
+        providerId: name-in-vault
+        provider: conjur
+    config:
+      authStrategy: aws
+      pattern: ^http.*
```

Comment:
We had actually changes `pattern` something that gives it the necessary context.  `pattern` just tells you that it's a regex, but a regex for what?  What is its purpose? 
 `requestMatching` or, perhaps even better,`applyToRequestsMatching` tells you what it's doing.

Yes, yes, it's too long!  It's ugly.  But hear me out for a sec...

While I would never agree to name a method `applyToRequestsMatching`, I want to argue that the yml file context is different.  We don't have to type it over and over, and we don't have to use it to reason about object interactions.  Our only real goal here is to make it fully clear to someone reading the file, so that they know what it is _without consulting the docs_.  

Open to improvement, of course, but I do think we should find something better than `pattern`.